File: ios/Sources/ForgeIOSRuntime/StableTabsState.swift
import Foundation

public enum StableTabsState {
    public static func selected(ids: [String], requested: String?, fallback: String?) -> String? {
        if let requested, ids.contains(requested) { return requested }
        if let fallback, ids.contains(fallback) { return fallback }
        return ids.first
    }
    public static func mounted(ids: [String], selected: String?, visited: Set<String>, keepVisited: Bool, activeOnly: Bool) -> [String] {
        if !activeOnly && !keepVisited { return ids }
        return ids.filter { $0 == selected || (keepVisited && visited.contains($0)) }
    }
}

File: android/sdk/src/main/java/com/viant/forgeandroid/runtime/StableTabsState.kt
package com.viant.forgeandroid.runtime

object StableTabsState {
    fun selected(ids: List<String>, requested: String?, fallback: String?): String? = requested?.takeIf(ids::contains) ?: fallback?.takeIf(ids::contains) ?: ids.firstOrNull()
    fun mounted(ids: List<String>, selected: String?, visited: Set<String>, keepVisited: Boolean, activeOnly: Boolean): List<String> =
        if (!activeOnly && !keepVisited) ids else ids.filter { it == selected || (keepVisited && it in visited) }
}
